Formula/mingw-w64.rb

```diff
@@ -0,0 +1,303 @@
+class MingwW64 < Formula
+  def arch
+    if Hardware::CPU.type == :intel
+      if MacOS.prefer_64_bit?
+        "x86_64"
+      else
+        "i686"
+      end
+    elsif Hardware::CPU.type == :ppc
+      if MacOS.prefer_64_bit?
+        "powerpc64"
+      else
+        "powerpc"
+      end
+    end
+  end
+
+  def osmajor
+    `uname -r`.chomp
+  end
+
+  def target_archs
+    archs = []
+    archs.push("i686-w64-mingw32") if build.with?("i686")
+    archs.push("x86_64-w64-mingw32") if build.with?("x86_64")
+    if build.without?("i686") && build.without?("x86_64")
+      odie "Options --without-i686 and --without-x86_64 are mutually exclusive"
+    end
+  end
+
+  desc "Minimalist GNU for Windows with GCC."
+  homepage "https://mingw-w64.org/"
+  # https://sourceforge.net/p/mingw-w64/wiki2/Cross%20Win32%20and%20Win64%20compiler/
+  url "https://downloads.sourceforge.net/project/mingw-w64/mingw-w64/mingw-w64-release/mingw-w64-v4.0.6.tar.bz2"
+  sha256 "0c407394b0d8635553f4fbca674cdfe446aac223e90b4010603d863e4bdd015c"
+
+  option "without-i686", "Compile without i686 compiler"
+  option "without-x86_64", "Compile without x86_64 compiler"
+  # option "without-multilib", "Compile x86_64 copmiler without multilib support"
+
+  depends_on "gcc" => :build
+  depends_on "texinfo" => :build
+  depends_on "mpfr"
+  depends_on "libmpc"
+  depends_on "cloog"
+  depends_on "wine" => :optional
```

Comment:
Is the wine dependency for testing only? If so, maybe we can check the produced files in ways other than running them (objdump, etc.)